api/drivers/aws/driver.js

```diff
@@ -498,6 +429,133 @@ class AWSDriver extends Driver {
+    return this.getServer(machine.id)
+      .then((res) => {
+        return new Promise((resolve, reject) => {
+          const server = res.server;
+          const password = res.password;
+          const ip = res.addresses.public[0] || server.addresses.private[0];
+          this._client.ec2.waitFor('instanceStatusOk', {
+            InstanceIds: [machine.id]
+          }, function(err, data) {
+            if (err) {
+              reject(err);
+            } else if (!data) {
+              machine.status = 'pending';
+            } else if (data.InstanceStatuses[0].InstanceState.Code === 16) {
+              machine.status = 'running';
+            } else {
+              machine.status = 'unknown';
+            }
+            machine.password = password;
+            machine.ip = ip;
+            return resolve(machine);
+          });
+        });
+      });
+  }
+
+  /**
+   * Retrieve the machine's password
+   *
+   * @method getPassword
+   * @param {machine} Machine model
+   * @return {Promise[String]}
+   */
+  getPassword(machine) {
+    var image = null;
+    var server = null;
+    return MachineService.getDefaultImage()
+      .then((res) => {
+        image = res;
+        return this.getServer(machine.id);
+      })
+      .then((res) => {
+        server = res;
+        return new Promise((resolve, reject) => {
+          if (image.password === null) {
+            return this._client.ec2
+              .waitFor('passwordDataAvailable', {
+                InstanceId: machine.id
+              }, (err, res) => {
+                if (err) {
+                  return reject(err);
+                }
+                return resolve({
+                  server: server,
+                  password: res.PasswordData,
+                  image: image
+                });
+              });
+          }
+          this._client.ec2
+            .waitFor('instanceStatusOk', {
+              InstanceIds: [machine.id]
+            }, (err) => {
+
+              if (err) {
+                return reject(err);
+              }
+
+              this._client.ec2
+                .waitFor('systemStatusOk', {
+                  InstanceIds: [machine.id]
+                }, (err) => {
+                  if (err) {
+                    return reject(err);
+                  }
+
+                  return resolve({
+                    server: server,
+                    password: image.password,
+                    image: image
+                  });
+                });
+            });
+        });
+      })
+      .then((res) => {
+        return new Promise((resolve, reject) => {
+          res.server.refresh((err, server) => {
+            if (err) {
+              reject(err);
+            } else {
+              res.server = server;
+              resolve(res);
+            }
+          });
+        });
+      })
+      .then((res) => {
+        const password = res.password;
+        let _retrievePassword = function(password) {
```

Comment:
I fail to see the purpose of this function.
Why not use `Promise.resolve()` and `Promise.reject` directly ?